packages/flutter/lib/src/material/search_anchor.dart

```diff
@@ -477,24 +483,44 @@ class _SearchViewRoute extends PopupRoute<_SearchViewRoute> {
     final Size screenSize = MediaQuery.of(context).size;
     final Rect anchorRect = getRect() ?? Rect.zero;
 
-    // Check if the search view goes off the screen.
     final BoxConstraints effectiveConstraints = viewConstraints ?? viewTheme.constraints ?? viewDefaults.constraints!;
-    final double verticalDistanceToEdge = screenSize.height - anchorRect.top;
-    final double endHeight = math.max(effectiveConstraints.minHeight, math.min(screenSize.height * 2 / 3, verticalDistanceToEdge));
     _rectTween.begin = anchorRect;
 
+    double viewWidth = anchorRect.width;
```

Comment:
It looks like much of the code here and in didChangeDependencies is at least very similar. In a future PR, factoring out a utility method that adjust the search view's bounds to fit, might be a good idea.

Reply:
I see. Will create a refactor PR to fix this and another [comment](https://github.com/flutter/flutter/pull/123256#discussion_r1148113121) in the previous PR. Thanks!